.github/workflows/release.yml

```diff
@@ -27,29 +32,28 @@ jobs:
           vault-role-id: ${{ secrets.VAULT_ROLE_ID }}
           vault-secret-id: ${{ secrets.VAULT_SECRET_ID }}
       - run: npm ci # runs npm prepublish
-      - name: Bump the version
-        run: |
-          echo "VERSION=$(npm version)" >> $GITHUB_ENV
-          echo "NPM_PACKAGE=$(jq '.name' package.json) >> $GITHUB_ENV
-      - run: git push --tags
       - uses: ./.github/actions/setup-totp
         with:
           vault-url: ${{ secrets.VAULT_ADDR }}
           vault-role-id: ${{ secrets.VAULT_ROLE_ID }}
           vault-secret-id: ${{ secrets.VAULT_SECRET_ID }}
-      - name: NPM Publish
-        run: npm publish --otp=${TOTP_CODE}
+      - run: npx semantic-release --dry-run="${DRY_RUN}"
+        env:
+          DRY_RUN: ${{ github.event.inputs.dry-run }}
+          GITHUB_TOKEN: ${{ github.token }}
+
+      - name: Get version and package name
+        run: |
+          echo "VERSION=$(jq -r '.version' package.json)" >> $GITHUB_ENV
+          echo "NPM_PACKAGE=$(jq -r '.name' package.json) >> $GITHUB_ENV
+
       - name: Create 'latest' dist-tag
         run: npm dist-tag --otp=${TOTP_CODE} add "${NPM_PACKAGE}@${VERSION}" latest
```

Comment:
if `dry_run` then do nothing? ditto with the remaining steps

Reply:
uh.. thank you. I missed that

Reply:
https://github.com/elastic/synthetics/pull/829/commits/6bd69f65398de6dc3e2be2f2b9dc540e8bc2fbbe